include/seastar/websocket/server.hh

```diff
@@ -159,33 +156,50 @@ public:
  */
 class connection : public boost::intrusive::list_base_hook<> {
     using buff_t = temporary_buffer<char>;
-    using writer_ptr = std::unique_ptr<pipe_writer<buff_t>>;
-    using reader_ptr = std::unique_ptr<pipe_reader<buff_t>>;
 
     /*!
      * \brief Implementation of connection's data source.
      */
     class connection_source_impl final : public data_source_impl {
-        reader_ptr pip;
+        queue<buff_t>* data;
```

Comment:
I will be easier to review if you squash this patch with the original one which introduced the usage of pipes in the first place.
In the first patch I had to wonder if pipe was the right implementation and whether you used it correctly (e.g., why the unique_ptr to the writer object) - and now suddenly it changes. Better just show the final decision.